crates/commitlog/src/commitlog.rs

```diff
@@ -137,62 +178,77 @@ impl<R: Repo, T> Generic<R, T> {
         Commits {
             inner: None,
             segments,
-            last_offset,
+            last_commit: CommitInfo::Initial { next_offset },
             last_error: None,
         }
     }
 
     pub fn reset(mut self) -> io::Result<Self> {
         info!("hard reset");
 
+        self.panicked = true;
```

Comment:
true or false?

Reply:
Yes, because we might have deleted / truncated the segment the buffered commits are supposed to be appended to.

NOTE TO SELF: maybe we could flush before resetting.